tests/Buildalyzer.Tests/Integration/SolutionWithProjectsFixture.cs

```diff
@@ -0,0 +1,116 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Buildalyzer.Environment;
+using NUnit.Framework;
+using Shouldly;
+
+namespace Buildalyzer.Tests.Integration
+{
+    [TestFixture]
+    [NonParallelizable]
+    public class SolutionWithProjectsFixture
+    {
+        private static readonly string[] SolutionFiles =
+        {
+            "NetCoreProjects.sln",
+            "NetStandardProjects.sln",
+            "NetCoreAndStandardProjects.sln"
+        };
+
+        [Test]
+        public void DesignTimeLoadsSolutionLogs(
+            [ValueSource(nameof(SolutionFiles))] string solutionFile)
+        {
+            // Given
+            StringWriter log = new StringWriter();
+            AnalyzerManager analyzerManager = GetAnalyzerManager(log);
+            EnvironmentOptions options = new EnvironmentOptions
+            {
+                Preference = EnvironmentPreference.Core
+            };
+            string binlogPath = GetBinlogPath(solutionFile);
+            string solutionPath = GetSolutionPath(solutionFile);
+
+            // When
+            /*
+            if (File.Exists(binlogPath))
+            {
+                File.Delete(binlogPath);
+            }
+            Dictionary<string, string> environmentVariables = new Dictionary<string, string>
+            {
+                { EnvironmentVariables.DOTNET_CLI_UI_LANGUAGE, "en-US" },
+                { EnvironmentVariables.MSBUILD_EXE_PATH, null },
+                { MsBuildProperties.MSBuildExtensionsPath, null }
+            };
+
+            using (ProcessRunner processRunner = new ProcessRunner("dotnet", $"clean {solutionPath}", Path.GetDirectoryName(solutionPath), environmentVariables, null))
+            {
+                processRunner.Start();
+                processRunner.WaitForExit(60000);
+            }
+
+            using (ProcessRunner processRunner = new ProcessRunner("dotnet", $"build {solutionPath} /bl:{binlogPath}", Path.GetDirectoryName(solutionPath), environmentVariables, null))
+            {
+                processRunner.Start();
+                processRunner.WaitForExit(60000);
+            }
+            */
```

Comment:
Any particular reason the binlog generation code here is commented out? Without it the tests obviously fail on a new checkout.

Reply:
Is a work in progress and will execute that code. I just checked this is for you to see where I am trying to go with the changes